altair/vegalite/v5/schema/_config.py

```diff
@@ -0,0 +1,1055 @@
+        | MarkConfigKwds
+        | RectConfigKwds
+        | TickConfigKwds
+    )
+    borders: (
+        bool
+        | BarConfigKwds
+        | AreaConfigKwds
+        | LineConfigKwds
+        | MarkConfigKwds
+        | RectConfigKwds
+        | TickConfigKwds
+    )
+    extent: ErrorBarExtent_T
+    interpolate: Interpolate_T
+    tension: float
+
+
+class ErrorBarConfigKwds(TypedDict, total=False):
+    """Placeholder doc."""
+
+    extent: ErrorBarExtent_T
+    rule: (
+        bool
+        | BarConfigKwds
+        | AreaConfigKwds
+        | LineConfigKwds
+        | MarkConfigKwds
+        | RectConfigKwds
+        | TickConfigKwds
+    )
+    size: float
+    thickness: float
+    ticks: (
+        bool
+        | BarConfigKwds
+        | AreaConfigKwds
+        | LineConfigKwds
+        | MarkConfigKwds
+        | RectConfigKwds
+        | TickConfigKwds
+    )
+
+
+class HeaderConfigKwds(TypedDict, total=False):
+    """Placeholder doc."""
+
+    format: str | Dict
+    formatType: str
+    labelAlign: Align_T
+    labelAnchor: TitleAnchor_T
+    labelAngle: float
+    labelBaseline: str | Baseline_T
+    labelColor: str | ColorName_T
+    labelExpr: str
+    labelFont: str
+    labelFontSize: float
+    labelFontStyle: str | FontStyle
+    labelFontWeight: FontWeight_T
+    labelLimit: float
+    labelLineHeight: float
+    labelOrient: Orient_T
+    labelPadding: float
+    labels: bool
+    orient: Orient_T
+    title: None
+    titleAlign: Align_T
+    titleAnchor: TitleAnchor_T
+    titleAngle: float
+    titleBaseline: str | Baseline_T
+    titleColor: str | ColorName_T
+    titleFont: str
+    titleFontSize: float
+    titleFontStyle: str | FontStyle
+    titleFontWeight: FontWeight_T
+    titleLimit: float
+    titleLineHeight: float
+    titleOrient: Orient_T
+    titlePadding: float
+
+
+class LegendConfigKwds(TypedDict, total=False):
+    """Placeholder doc."""
+
+    aria: bool
+    clipHeight: float
+    columnPadding: float
+    columns: float
+    cornerRadius: float
+    description: str
+    direction: Orientation_T
+    disable: bool
+    fillColor: str | None | ColorName_T
+    gradientDirection: Orientation_T
+    gradientHorizontalMaxLength: float
+    gradientHorizontalMinLength: float
+    gradientLabelLimit: float
+    gradientLabelOffset: float
+    gradientLength: float
+    gradientOpacity: float
+    gradientStrokeColor: str | None | ColorName_T
+    gradientStrokeWidth: float
+    gradientThickness: float
+    gradientVerticalMaxLength: float
+    gradientVerticalMinLength: float
+    gridAlign: LayoutAlign_T
+    labelAlign: Align_T
+    labelBaseline: str | Baseline_T
+    labelColor: str | None | ColorName_T
+    labelFont: str
+    labelFontSize: float
+    labelFontStyle: str | FontStyle
+    labelFontWeight: FontWeight_T
+    labelLimit: float
+    labelOffset: float
+    labelOpacity: float
+    labelOverlap: str | bool
+    labelPadding: float
+    labelSeparation: float
+    layout: Map
```

Comment:
HACK: `Map` is being used since it would be `dict | ExprRef`

Reply:
I think this might not be relevant anymore since `Map` is used everywhere now instead of `dict`
- https://github.com/vega/altair/pull/3573

Reply:
https://altair-viz.github.io/user_guide/generated/core/altair.LegendConfig.html#altair.LegendConfig

`LegendConfig.layout` seems to not have a description?